src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -127,6 +127,14 @@ private async Task ProcessAsyncActions(AsyncAction[][] actions)
                         case AsyncActionType.CreateTimer:
                             using (var cts = new CancellationTokenSource())
                             {
+                                DurableOrchestrationContext ctx = this.context as DurableOrchestrationContext;
```

Comment:
We should handle the case where ctx is null (i.e. where this.context is not of type `DurableOrchestrationContext`). We can do this by throwing an exception.

This should only happen in test code where we pass in a mock IDurableOrchestrationContext, but getting a nice exception as opposed to a null ref exception is always preferred.